packages/bitcore-node/src/models/transaction.ts

```diff
@@ -404,15 +404,24 @@ export class TransactionModel extends BaseModel<ITransaction> {
 
     const walletConfig = Config.for('api').wallets;
     if (initialSyncComplete || (walletConfig && walletConfig.allowCreationBeforeCompleteSync)) {
-      let mintOpsAddresses = {};
-      for (const mintOp of mintOps) {
-        mintOpsAddresses[mintOp.updateOne.update.$set.address] = true;
-      }
-      mintOpsAddresses = Object.keys(mintOpsAddresses);
-      let wallets = await WalletAddressStorage.collection
-        .find({ address: { $in: mintOpsAddresses }, chain, network }, { batchSize: 100 })
-        .project({ wallet: 1, address: 1 })
-        .toArray();
+      let mintOpsAddressesSet = {};
+      for (const mintOp of mintOps) { mintOpsAddressesSet[mintOp.updateOne.update.$set.address] = true; }
+      var mintOpsAddresses = Object.keys(mintOpsAddressesSet);
+
+      let wallets : IWalletAddress[] = [];
+
+      await Promise.all(
+        partition(mintOpsAddresses, mintOpsAddresses.length / Config.get().maxPoolSize).map(async addressesBatch => 
```

Comment:
What is the value of `Config.get().maxPoolSize` ? This constant may be used for other things and not necessarily make sense in this context? Probably it makes sense to just hard code a number here instead or ask when you make the PR upstream if there is a constant that should be used or if a new constant can be created for this case.

Reply:
Good point, this constant is set by default to `50` here:
https://github.com/cwcrypto/bitcore-1/blob/469dfbbb41708ee14e4f4a8ac3426a8611f068bf/packages/bitcore-node/src/config.ts#L55-L56

I used this constant just following how I've seen this approach done in other places of the code, like here:
https://github.com/cwcrypto/bitcore-1/blob/61ddc54b045f7eca12b4fdeca6a882ab58d94ca6/packages/bitcore-node/src/models/transaction.ts#L142-L143

I'm still not sure what's the max buffer size that `mongo` accepts, because I'm worried that maybe in some cases (extreme cases that won't happen now) that even after partitioning, the resulting partition is still big enough to cause that error.

I Will definitely ask about it when I make the PR to upstream, but I think this might be good enough for now?

Reply:
Ok yeah that works for now. Yeah it really makes more sense to have a mongo query limit constant or something so this can apply to all mongo queries that are made that may need to filter on many values but decision should definitely be made upstream.